MetaMask/metamask-extension
# [DARKMODE] Initial new color design tokens and docs

Fixes: #13564

Explanation:  Adding new design token color style sheets and docs. This is the first step towards theming and dark mode. The next step would be to swap out all of the static hex and current color variables to the ones in this PR.

**Notes**
- Values of the tokens may be subject to change. This should only "tweak" the colors and shouldn't effect how they are to be used throughout the codebase.
- Using HSLA color mode values as we thought it might help consistency with the transparent values and in my opinion is  superior and most logical and of the colormodes. Might be better to use hex values until we adopt HSLA as a company.. I did only find out that you can do alpha with HEX values now so those could be updated too.
- Ideally we should be using the [`design-tokens`](https://github.com/MetaMask/design-tokens) package to generate these stylesheets but for MVP and to help with automation we could start using the end result and then hook everything up to be automated.
- Would be nice to auto generate the documentation based on the design-tokens repo

**TO DO**
- [ ] Update `ui/helpers/constants/design-system.js` with all the new colors

Manual testing steps:  
  - Run `yarn storybook`
  - Go to color docs at http://localhost:6006/?path=/story/design-tokens-color-getting-started--page
  - Read docs and check against code
 
## Screenshots

![colors getting started](https://user-images.githubusercontent.com/8112138/153323916-e69b8044-137f-49ae-9422-2e29f2319b9c.png)
<img width="1439" alt="Screen Shot 2022-02-10 at 12 37 05 PM" src="https://user-images.githubusercontent.com/8112138/153324248-6d1a231c-7e3d-4d67-ad71-aaeb11d4e496.png">
<img width="1440" alt="Screen Shot 2022-02-10 at 12 37 17 PM" src="https://user-images.githubusercontent.com/8112138/153324264-0a533f25-006c-4fd9-9c89-d52355ba51e7.png">
